x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/10_basic.yml

```diff
@@ -309,12 +309,15 @@ setup:
   - match: {values.0: [1, 44, "green"]}
 
 ---
-"Test Mixed Input Params":
+"Test Named Input Params":
+  - requires:
+      cluster_features: ["gte_v8.15.0"]
+      reason: "Support named parameters in 8.15+"
   - do:
       esql.query:
         body:
           query: 'from test | eval x = ?, y = ?, z = ?, t = ?, u = ?, v = ? | keep x, y, z, t, u, v | limit 3'
-          params: [{"value": 1, "type": "keyword"}, {"value": 2, "type": "double"}, null, true, 123, {"value": 123, "type": "long"}]
+          params: [{"n1" : "1"}, {"n2" : 2.0}, {"n3" : null}, {"n4" : true}, {"n5" : 123}, {"n6": 1674835275193}]
```

Comment:
Please add a test that preserves the original test with unnamed params (without type) like the original one.